fda/fdata_boxplot.py

```diff
@@ -0,0 +1,627 @@
+
+                    # outlying envelope
+                    max_value = np.amax(fdatagrid.data_matrix[:, :, m], axis=0)
+                    min_value = np.amin(fdatagrid.data_matrix[:, :, m], axis=0)
+                    iqr = np.absolute(max_samples_used - min_samples_used)
+                    outlying_max_envelope = np.minimum(
+                        max_samples_used + iqr * factor, max_value)
+                    outlying_min_envelope = np.maximum(
+                        min_samples_used - iqr * factor, min_value)
+                    self._outlying_envelope[m] = np.asarray(
+                        [outlying_max_envelope.flatten(),
+                         outlying_min_envelope.flatten()])
+
+                    # outliers
+                    for j in list(range(fdatagrid.nsamples)):
+                        outliers_above = (
+                                outlying_max_envelope < fdatagrid.data_matrix[
+                                                        j, :, m])
+                        outliers_below = (
+                                outlying_min_envelope > fdatagrid.data_matrix[
+                                                        j, :, m])
+                        if (
+                                outliers_above.sum() > 0 or outliers_below.sum() > 0):
+                            self._outliers[m, j] = 1
+                # central regions
+                self._central_regions[ncentral_regions * m + i] = np.asarray(
+                    [max_samples_used.flatten(), min_samples_used.flatten()])
+
+            # mean sample
+            self._median[m] = fdatagrid.data_matrix[
+                              indices_descencing_depth[0, m], :, m].T
+
+        self._fdatagrid = fdatagrid
+        self._prob = prob
+        self._colormap = plt.cm.get_cmap('RdPu')
+        self._barcol = "blue"
+        self._outliercol = "red"
+        self._mediancol = "black"
+        self._fullout = False
+
+    @property
+    def fdatagrid(self):
+        return self._fdatagrid
+
+    @property
+    def median(self):
+        return self._median
+
+    @property
+    def central_envelope(self):
+        return self._central_envelope
+
+    @property
+    def outlying_envelope(self):
+        return self._outlying_envelope
+
+    @property
+    def colormap(self):
+        return self._colormap
+
+    @colormap.setter
+    def colormap(self, value):
+        if not isinstance(value, matplotlib.colors.LinearSegmentedColormap):
+            raise ValueError(
+                "colormap must be of type matplotlib.colors.LinearSegmentedColormap")
+        self._colormap = value
+
+    @property
+    def central_regions(self):
+        return self._central_regions
+
+    @property
+    def outliers(self):
+        return self._outliers
+
+    @property
+    def barcol(self):
+        return self._barcol
+
+    @barcol.setter
+    def barcol(self, color):
+        self._barcol = color
```

Comment:
If the getter and setter are not doing additional work, is preferable to remove them and expose a public attribute (in Python).

Reply:
Do you mean to initialize the attribute in the _init_ function as self.barcol = "blue" and remove both 
setter and getter?

Reply:
Yes.